src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -375,12 +375,20 @@ loop_inst::concatenated_memory_mapping::ptr loop_inst::create_concat_memory_map(
     if (extern_mem_ptr != nullptr) {
         layout sliced_layout = intern_prim->get_output_layout(internal_id.idx);
         auto inter_mem_ptr = intern_prim->output_memory_ptr(internal_id.idx);
-        if (inter_mem_ptr == nullptr) {
+        if (inter_mem_ptr == nullptr || shape_changed()) {
             // if inner body intern_prim has no output memory because it has dynamic shape,
             // calculate inner body intern_prim layout using concat_mem's layout.
             auto updated_sliced_layout = sliced_layout.get_partial_shape();
             OPENVINO_ASSERT(updated_sliced_layout[io_prim_map.axis].is_static() || num_iterations > 0,
                                     "Not allowed dynamic dimension for axis when num_iteraiont is negative");
+
+            // Save or load dynamic sliced layout to update with various shapes
```

Comment:
If a loop has multiple sliced inputs, how can you know this dynamic_sliced_layout is related to which sliced input?
I think it is better to handle dynamic layout with containers such as std::vector or std::map.

Reply:
Applied